### PR TITLE
chore: soft-block commits to main branch with husky pre-commit check

### LINKED ISSUE #11091: Use automation to soft-block commits to `main` branch
**What would you like Renovate to be able to do?**

Our [development docs warn contributors to not commit to their forks `main` branch](https://github.com/renovatebot/renovate/blob/main/docs/development/local-development.md#keeping-your-renovate-fork-up-to-date), but it can be very easy to do this by accident anyway. 😄 

It would be nice to have the computer check for the bad branch, and soft-block you from committing.

Note that this setup would make life a bit harder for @rarkins, in case they want to commit to `main` directly without a PR!

**Did you already have any implementation ideas?**

I think we can use Husky with a pre-commit hook to check what branch we're on, and soft-block commits to the `main` branch.

Workflow would be like this:

1. User checks out `main` branch
2. User makes changes to `main` branch by accident
3. User commits
4. Husky runs check for `main` branch.
5. Husky complains: "You're on the `main` branch, are you sure you want to commit?"
6. Abort commit, so that user can do `git switch -c feature-branch/issue-1`
7. Optional: provide option to commit to `main` branch anyways by passing a flag to Husky.

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes:

- Add husky pre-commit check to soft-block commits to `main` branch

## Context:

Marking this as a draft PR until we're happy with the code.

Fixes #11091.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — DISMISSED** [review]

**Comment by viceice** [comment]
Works for me, if i change the the branch name to `pr/HonkingGoose/11094` 😉

**Comment by rarkins** [comment]
Let's try it

File: .husky/pre-commit [hunk 1]
```diff
@@ -2,3 +2,22 @@
 . "$(dirname "$0")/_/husky.sh"
 
 yarn pretty-quick --staged
+
+BRANCH_NAME=$(git branch --show-current)
+
+# I'm using "chore/husky-git-hooks" so that you can test my work in the PR branch
+# TODO: Replace "chore/husky-git-hooks" with "main" in the IF statement, and remove both comments before marking this "ready to merge" on GitHub
+
+if [ "$BRANCH_NAME" = "chore/husky-git-hooks" ]; then
```
Comment (viceice) [inline .husky/pre-commit:11]:
```suggestion
if [ "$BRANCH_NAME" = "main" ]; then
```

**Review by viceice — APPROVED** [review]

**Comment by viceice** [comment]
@rarkins This hook will only run after yarn deps are installed. So it mostly will only be visible by developers.

**Pull request merged**